lib/js/emojione.js

```diff
@@ -425,13 +448,13 @@
         return string.replace(/[-[\]{}()*+?.,;:&\\^$|#\s]/g, "\\$&")
     }
 
-    ns.replaceAll = function(string, find, replaceWith) {
+    ns.replaceAll = function(string, find, replacementList) {
 
         var search = new RegExp("<object[^>]*>.*?<\/object>|<span[^>]*>.*?<\/span>|<(?:object|embed|svg|img|div|span|p|a)[^>]*>|("+find+")", "gi");
 
         // callback prevents replacing anything inside of these common html tags as well as between an <object></object> tag
         var replace = function(entire, m1) {
-            return ((m1 == undefined) || (m1 == '')) ? entire : replaceWith;
+            return replacementList[m1];
```

Comment:
Maybe I was too optimistic not checking for emptiness. Now I'm thinking to replace it for something like:

    return replacementList[m1] || m1;